src/components/views/rooms/AppMaximisedDrawer.tsx

```diff
@@ -0,0 +1,143 @@
+/*
+Copyright 2017 Vector Creations Ltd
+Copyright 2018 New Vector Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from 'react';
+
+import AppTile from '../elements/AppTile';
+import dis from '../../../dispatcher/dispatcher';
+import * as sdk from '../../../index';
+import * as ScalarMessaging from '../../../ScalarMessaging';
+import WidgetUtils from '../../../utils/WidgetUtils';
+import WidgetEchoStore from "../../../stores/WidgetEchoStore";
+import ResizeNotifier from "../../../utils/ResizeNotifier";
+import { Container, WidgetLayoutStore } from "../../../stores/widgets/WidgetLayoutStore";
+import { replaceableComponent } from "../../../utils/replaceableComponent";
+import { Room } from "matrix-js-sdk/src/models/room";
+import { IApp } from "../../../stores/WidgetStore";
+
+interface IProps {
+    userId: string;
+    room: Room;
+    resizeNotifier: ResizeNotifier;
+    showApps?: boolean; // Should apps be rendered
+    maxHeight: number;
+}
+
+interface IState {
+    app: IApp;
+    resizing: boolean;
+}
+
+@replaceableComponent("views.rooms.AppMaximisedDrawer")
+export default class AppMaximisedDrawer extends React.Component<IProps, IState> {
```

Comment:
Hmm, so a lot of this is copied from `AppsDrawer` I assume... Is it possible to share more instead? It feels like a lot of duplication at the moment.

Reply:
I was thinking about this as well. Another option is to extend the appDrawer with a maximised property. I will investigate that option as well. I thought there might be more special cases for the maximised drawer but it turned out super simple, so maybe extending the AppDrawer is the better choice after all.

Reply:
Now we reuse the apps drawer.